xbmc/addons/AddonBuilder.cpp

```diff
@@ -81,65 +116,26 @@ std::shared_ptr<IAddon> CAddonBuilder::Build()
+      return std::make_shared<CScreenSaver>(std::move(m_props));
+#ifdef HAS_PVRCLIENTS
     case ADDON_PVRDLL:
+      return PVR::CPVRClient::FromExtension(std::move(m_props), m_extPoint);
+#endif
     case ADDON_ADSPDLL:
+      return std::make_shared<ActiveAE::CActiveAEDSPAddon>(std::move(m_props));
     case ADDON_AUDIOENCODER:
+      return CAudioEncoder::FromExtension(std::move(m_props), m_extPoint);
     case ADDON_AUDIODECODER:
+      return CAudioDecoder::FromExtension(std::move(m_props), m_extPoint);
     case ADDON_INPUTSTREAM:
+      return CInputStream::FromExtension(std::move(m_props), m_extPoint);
     case ADDON_PERIPHERALDLL:
-    { // begin temporary platform handling for Dlls
-      // ideally platforms issues will be handled by C-Pluff
-      // this is not an attempt at a solution
-      std::string value;
-      if (type == ADDON_SCREENSAVER && 0 == strnicmp(m_extPoint->plugin->identifier, "screensaver.xbmc.builtin.", 25))
-      { // built in screensaver
-        return std::make_shared<CAddon>(std::move(m_props));
-      }
-      if (type == ADDON_SCREENSAVER)
-      { // Python screensaver
-        std::string library = CAddonMgr::GetInstance().GetExtValue(m_extPoint->configuration, "@library");
-        if (URIUtils::HasExtension(library, ".py"))
-          return std::make_shared<CScreenSaver>(std::move(m_props));
-      }
-      if (type == ADDON_AUDIOENCODER && 0 == strncmp(m_extPoint->plugin->identifier,
-          "audioencoder.xbmc.builtin.", 26))
-      { // built in audio encoder
-        return CAudioEncoder::FromExtension(std::move(m_props), m_extPoint);
-      }
-
-      value = CAddonMgr::GetInstance().GetPlatformLibraryName(m_extPoint->plugin->extensions->configuration);
-      if (value.empty())
-        break;
-      if (type == ADDON_VIZ)
-      {
-#if defined(HAS_VISUALISATION)
-        return std::make_shared<CVisualisation>(std::move(m_props));
-#endif
-      }
-      else if (type == ADDON_PVRDLL)
-      {
-#ifdef HAS_PVRCLIENTS
-        return PVR::CPVRClient::FromExtension(std::move(m_props), m_extPoint);
-#endif
-      }
-      else if (type == ADDON_ADSPDLL)
-      {
-        return std::make_shared<ActiveAE::CActiveAEDSPAddon>(std::move(m_props));
-      }
-      else if (type == ADDON_AUDIOENCODER)
-        return CAudioEncoder::FromExtension(std::move(m_props), m_extPoint);
-      else if (type == ADDON_AUDIODECODER)
-        return CAudioDecoder::FromExtension(std::move(m_props), m_extPoint);
-      else if (type == ADDON_INPUTSTREAM)
-        return CInputStream::FromExtension(std::move(m_props), m_extPoint);
-      else if (type == ADDON_SCREENSAVER)
-        return std::make_shared<CScreenSaver>(std::move(m_props));
-      else if (type == ADDON_PERIPHERALDLL)
-        return PERIPHERALS::CPeripheralAddon::FromExtension(std::move(m_props), m_extPoint);
-      break;
-    }
+      return PERIPHERALS::CPeripheralAddon::FromExtension(std::move(m_props), m_extPoint);
```

Comment:
Correct, the coverty report made no sense. Martijn received an extended report with the words "must be true", which clued me into the defect. We have code like:

```c++
switch (foo) 
{
case A:
case B:
case C:
  if (A) ...
  else if (B) ...
  else if (C) ...
  break;
}
```
OFC, the `break` can never be hit, because C always "must be true".

This refactors the special casing to above the switch and simplifies the switch.